src/main/java/rx/internal/util/ScalarSynchronousObservable.java

```diff
@@ -29,28 +34,105 @@
     private final T t;
 
     protected ScalarSynchronousObservable(final T t) {
-        super(new OnSubscribe<T>() {
+        super(new ScalarSynchronousObservableOnSubscribe<T>(t));
+        this.t = t;
+    }
+    private static final class ScalarSynchronousObservableOnSubscribe<T> implements OnSubscribe<T> {
```

Comment:
If not than the debug hooks have gaps in the data flow.  It looks like `onNext`s are produced out of nowhere.